lms/static/scripts/frontend_apps/components/FilePickerApp.js

```diff
@@ -94,13 +98,39 @@ export default function FilePickerApp({ onSubmit }) {
   // Submit the form after a selection is made via one of the available
   // methods.
   useEffect(() => {
+    async function createAssignment() {
+      /** @type Record<string, string|Content|null> */
```

Comment:
Maybe just `{Object}`? I was confused but the error in the next line, see comment.

Reply:
If you create the `data` object in one step, you can avoid needing any type hints:

```diff
diff --git a/lms/static/scripts/frontend_apps/components/FilePickerApp.js b/lms/static/scripts/frontend_apps/components/FilePickerApp.js
index d1a94aae..8282b067 100644
--- a/lms/static/scripts/frontend_apps/components/FilePickerApp.js
+++ b/lms/static/scripts/frontend_apps/components/FilePickerApp.js
@@ -99,15 +99,16 @@ export default function FilePickerApp({ onSubmit }) {
   // methods.
   useEffect(() => {
     async function createAssignment() {
-      /** @type Record<string, string|Content|null> */
-      const data = { ...createAssignmentAPI.data };
+      const data = {
+        ...createAssignmentAPI.data,
+        content,
+        groupset: groupConfig.groupSet,
+      };

-      data.content = content;
-      data.groupset = groupConfig.groupSet;
       const assignment = await apiCall({
         authToken,
         path: createAssignmentAPI.path,
-        data: data,
+        data,
       });
       setExtLTIAssignmentId(assignment.ext_lti_assignment_id);
     }
```